cpp/tests/io/parquet_reader_test.cpp

```diff
@@ -2724,3 +2724,38 @@ TYPED_TEST(ParquetReaderPredicatePushdownTest, FilterTyped)
   EXPECT_EQ(result_table.num_columns(), expected->num_columns());
   CUDF_TEST_EXPECT_TABLES_EQUAL(expected->view(), result_table);
 }
+
+TEST_F(ParquetReaderTest, ListsWideTable)
+{
+  auto constexpr num_rows = 5;
+  auto constexpr num_cols = 32'768;
```

Comment:
how do we reach >2B keys in this test?

Reply:
The `num_keys` is calculated as `ncols x max_depth x total pages`. In this test we have 32768 cols x 2 max depth x 32768 pages = 2B exact

Reply:
can we make a smaller/simple table? AFAICT, we need `sqrt(max_keys)+1` columns, which can be simple int columns with a few values.

Reply:
I would say we make `num_cols = 32769` to slightly exceed 2B keys and really test the fix.

Reply:
> can we make a smaller/simple table? AFAICT, we need sqrt(int32::max)+1 columns, which can be simple int columns with a few values.

Or now that we have the fix, I guess we can make table smaller to have slightly over 1B keys with lists of ints maybe. We do need list cols though.

Reply:
We do need over 2B to have the pre-fix code fail, no?
Can we have a list of ints? Or whatever is the smallest repro table.

Reply:
Yep, so for 2B keys, we do need `num_cols=32769`. Let me make it lists of ints though.

Reply:
Converted to `list<list<int>>` requiring 26755 cols instead of 32769 for `list<int>`

Reply:
did this reduce the test execution time?

Reply:
>did this reduce the test execution time?

By roughly 1-2 seconds on my machine (9-10s compared to 10-12s with 32769 cols)